doc/api/inspector.md

```diff
@@ -0,0 +1,80 @@
+# Inspector
+
+> Stability: 1 - Experimental
+
+The `inspector` module provides an API for interacting with the V8 inspector.
+
+It can be accessed using:
+
+```js
+const inspector = require('inspector');
+```
+
+## Class: inspector.Session
+<!-- YAML
+added: REPLACEME
+-->
+
+The `inspector.Session` class is a primary means of dispatching the messages to
+the V8 inspector backend and receiving message responses and notifications.
+
+It needs to be connected before the messages can be sent to the inspector
+backend.
+
+### Event 'inspectorNotification'
+<!-- YAML
+added: REPLACEME
+-->
+
+* {Object} The notification
+
+Emitted when a notification from an inspector backend was received.
+
+It is also possible to subscribe to a specific inspector notification type by
+specifying the method. E.g. this snippet will print the reason for application
+been suspended when the [Debugger.paused][debugger-paused] event is recieved
+be called whenever the program execution is suspended:
+```js
+session.on('Debugger.paused', ({reason}) => console.log(reason));
+```
+
+### session.connect()
+<!-- YAML
+added: REPLACEME
+-->
+
+Connects a session to inspector backend. An exception will be thrown
+if a session is created when there is already an ongoing inspector session
+created either by using this API or by connecting to a WebSocket port.
+
+### session.post(method, params[, callback])
```

Comment:
It would be ideal if `params` was optional so that methods that take no params can receive a callback without needing to specify an extra `null` parameter.

The pattern elsewhere in Node is to check if what was passed as the second arg is in fact a Function (rather than an object) and if so assign it to `callback` instead. If that makes sense to you I think we should do that here.